modules/Core/pages/panel/users_reports.php

```diff
@@ -285,24 +286,29 @@
                 // Get report
                 $report = $queries->getWhere('reports', array('id', '=', $_GET['id']));
                 if (count($report)) {
-                    $queries->update('reports', $report[0]->id, array(
-                        'status' => 1,
-                        'date_updated' => date('Y-m-d H:i:s'),
-                        'updated' => date('U'),
-                        'updated_by' => $user->data()->id
-                    ));
-
-                    $queries->create('reports_comments', array(
-                        'report_id' => $report[0]->id,
-                        'commenter_id' => $user->data()->id,
-                        'comment_date' => date('Y-m-d H:i:s'),
-                        'date' => date('U'),
-                        'comment_content' => str_replace('{x}', Output::getClean($user->data()->username), $language->get('moderator', 'x_closed_report'))
-                    ));
+                    if($report[0]->status == 0){
+                        $queries->update('reports', $report[0]->id, array(
+                            'status' => 1,
+                            'date_updated' => date('Y-m-d H:i:s'),
+                            'updated' => date('U'),
+                            'updated_by' => $user->data()->id
+                        ));
+
+                        $queries->create('reports_comments', array(
+                            'report_id' => $report[0]->id,
+                            'commenter_id' => $user->data()->id,
+                            'comment_date' => date('Y-m-d H:i:s'),
+                            'date' => date('U'),
+                            'comment_content' => str_replace('{x}', Output::getClean($user->data()->username), $language->get('moderator', 'x_closed_report'))
+                        ));
+                        Session::flash('report_success', $language->get('moderator', 'report_closed'));
+                        Redirect::to(URL::build('/panel/users/reports/', 'id=' . Output::getClean($report[0]->id)));
+                    } else {
+
+                        Session::flash('report_failure', "You cannot close this report 2x");
```

Comment:
Could you add a translation for this, just like the other Session::flash lines?

Reply:
> Could you add a translation for this, just like the other Session::flash lines?

Done.

PS: Sorry for a large number of commits 😅

Reply:
don't worry, fortunately github has a squash option